wolox-books/src/views/Register.vue

```diff
@@ -0,0 +1,162 @@
+<template lang='pug'>
+  .container
+      .container-register
+        img.logo(alt='Wolox logo' src='../assets/logo.png')
+        form(@submit.prevent='onSubmit').container-form
+          .container-field(v-for='(field, name) in fields' :key='name')
+            label.label(:for='name')
+              | {{field.label}}
+            input.input(:type='field.type' v-model='field.value' :id='name' :class="{'error-input': field.validation.submitError}")
+            .error(v-if='field.validation.submitError') {{ field.validation.error }}
+          button.button-sign-up(:type='button') Sign Up
+        button.button-login(:type='submit') Login
+</template>
+<script>
+
+import { email, numeric, helpers } from 'vuelidate/lib/validators'
+import { validationPassword } from '../validations/inputValidations';
+import { errorPass, errorEmail } from '../validations/constants';
+
+export default {
+  name: 'Register',
+  data () {
+    return {
+      fields: {
+        firstName: { 
+          label: 'First name',
+          value: null ,
+          validation: {}
+        },
+        lastName: {
+          label: 'Last name',
+          value: null,
+          validation: {}
+        },
+        email: { 
+          label: 'Email',
+          value: null,
+          validation: {
+            submitError: false,
+            error: errorEmail
+          }
+        },
+        password: {
+          label: 'Password',
+          value: null,
+          type: 'password', 
+          validation: {
+            submitError: false,
+            error: errorPass
+          }
+        }
+      },
+      errors: []
+    }
+  },
+  validations: {
+    fields: {
+      email: {
+        value: { email }
+      },
+      password: {
+        value: { validationPassword }
+      }
+    }
+  },
+  methods: {
+    validationsFields() {
+      this.fields.email.validation.submitError = this.$v.fields.email.$invalid
+      this.fields.password.validation.submitError = this.$v.fields.password.$invalid
```

Comment:
You can just reference this in the template instead of doing this:

```pug
.error(v-if='$v.fields.email.$invalid')
  |  {{ email.error }}
```

That way there's no need to keep multiple variables for the same thing 😄